health/probe/plugin_impl_prometheus.go

```diff
@@ -121,60 +121,26 @@ func (p *PrometheusPlugin) AfterInit() error {
 		p.Log.Info("Unable to register Prometheus metrics handlers, HTTP is nil")
 	}
 
-	//TODO: Need improvement - instead of the exposing the map directly need to use in-memory mapping
-	if p.PluginStatusCheck != nil {
-		allPluginStatusMap := p.PluginStatusCheck.GetAllPluginStatus()
-		for k, v := range allPluginStatusMap {
-			p.Log.Infof("k=%v, v=%v, state=%v", k, v, v.State)
+	if p.StatusCheck != nil {
+		pluginStatusIdx := p.StatusCheck.GetAllPluginStatus()
+		allPluginNames := pluginStatusIdx.GetMapping().ListAllNames()
+		for _, v := range allPluginNames {
 			p.registerGauge(
 				Namespace,
 				Subsystem,
 				DependencyHealthName,
 				DependencyHealthHelp,
 				prometheus.Labels{
 					ServiceLabel:    p.getServiceLabel(),
-					DependencyLabel: k,
+					DependencyLabel: v,
 				},
-				p.getDependencyHealth(k, v),
+				p.getDependencyHealth(v, pluginStatusIdx),
```

Comment:
statuscheck allows to notify when plugin status changes using watch (instead of polling in getDependencyHealth) is it technically possible with prometheus?

Reply:
Would be good if we could have a quick discussion.